rust_src/src/frame.rs

```diff
@@ -691,4 +691,47 @@ pub extern "C" fn get_frame_param(frame: LispFrameRef, prop: LispObject) -> Lisp
     frame.get_param(prop)
 }
 
-include!(concat!(env!("OUT_DIR"), "/frames_exports.rs"));
+/// Height in pixels of a line in the font in frame FRAME.
+/// If FRAME is omitted or nil, the selected frame is used.
+/// For a terminal frame, the value is always 1.
+#[lisp_fn(min = "0")]
+pub fn frame_char_height(_frame: LispFrameLiveOrSelected) -> i32 {
+    #[cfg(feature = "window-system")]
+    {
+        let frame_ref: LispFrameRef = _frame.into();
+
+        if frame_ref.is_gui_window() {
+            frame_ref.line_height
+        } else {
+            1
+        }
+    }
+    #[cfg(not(feature = "window-system"))]
+    {
+        1
+    }
+}
+
```

Comment:
I'd recommend compressing this a bit if by changing the body of the if to `return frame_ref.line_height` and dropped the else and the second cfg, i.e.:
```rust
        if frame_ref.is_gui_window() {
            return frame_ref.line_height;
        }
    }
    1
}
```

`frame_char_width` can be shortened in a similar way.

Reply:
@agraven Thanks for the suggestion - that's a good idea.I'll update the PR when I get a chance.

The module is being renamed because it doesn't match the c file counterpart like the rest of the files do (and should). shaleh's suggestion was specifically regarding testing this change when in GUI mode.

Reply:
Ah, apologies, I misunderstood.

Reply:
@agraven To take your suggestion further, shouldn't it be possible to completely drop all the `cfg`s? It seems to me that `is_gui_window()` would simply return false when there is no window system, as it would be in terminal mode (or initial, whatever that is).

Reply:
That's true, I forgot that function wasn't behind a `cfg`. And I suspect initial is before either the terminal or the gui interface is initialized, but I'm not entirely sure.

Reply:
@aclonegeek that sounds reasonable, you might need to touch some other code.

Reply:
The main reason behind the C #ifdef logic is it is faster and yields a smaller binary. There is no reason to interrogate the frame if there is no GUI support.

Reply:
@shaleh Good point. I think we should keep `cfg` then, but only one of them is necessary (in this context). I'll apply @agraven's suggestion.